micromanager_gui/_core_widgets/_presets_widget.py

```diff
@@ -0,0 +1,139 @@
+from typing import Optional, Tuple
+
+from pymmcore_plus import DeviceType
+from qtpy.QtWidgets import QComboBox, QHBoxLayout, QWidget
+from superqt.utils import signals_blocked
+
+from .._core import get_core_singleton
+from .._util import get_dev_prop, get_dev_prop_val, set_wdg_color
+
+
+class PresetsWidget(QWidget):
+    """Create a QCombobox Widget for a specified group presets"""
+
+    def __init__(
+        self,
+        group: str,
+        text_color: str = "black",
+        parent: Optional[QWidget] = None,
+    ) -> None:
+
+        super().__init__(parent)
+
+        self._mmc = get_core_singleton()
+
+        self._group = group
+        self.text_color = text_color
+
+        if self._group not in self._mmc.getAvailableConfigGroups():
+            raise ValueError(f"{self._group} group does not exist.")
+
+        self._presets = list(self._mmc.getAvailableConfigs(self._group))
+
+        if not self._presets:
+            raise ValueError(f"{self._group} group does not have presets.")
+
+        self.dev_prop = get_dev_prop(self._group, self._presets[0])
```

Comment:
Ideally should check all presets. e.g. consider this (pathological) config group:

```
# Configuration presets
# Group: Camera
# Preset: MedRes
ConfigGroup,Camera,MedRes,Camera,Binning,2

# Preset: HighRes
ConfigGroup,Camera,HighRes,Camera,Binning,1
ConfigGroup,Camera,HighRes,Camera,BitDepth,12

# Preset: LowRes
ConfigGroup,Camera,LowRes,Camera,Binning,4
```

where bitdepth is in only one of the presets.

Reply:
@ianhi Do you mean that we should make sure that the each preset (from the same group) have the same (dev, prop)?

Like if the group `Camera` has 3 presets (`HighRes`, `MedRes`, `LowRes`), they all need to have the same properties (with different values):

```
# Group: Camera
# Preset: MedRes
ConfigGroup,Camera,MedRes,Camera,Binning,2

# Preset: HighRes
ConfigGroup,Camera,HighRes,Camera,Binning,1

# Preset: LowRes
ConfigGroup,Camera,LowRes,Camera,Binning,4
```

or 

```
# Group: Camera
# Preset: MedRes
ConfigGroup,Camera,MedRes,Camera,Binning,2
ConfigGroup,Camera,HighRes,Camera,BitDepth,10

# Preset: HighRes
ConfigGroup,Camera,HighRes,Camera,Binning,1
ConfigGroup,Camera,HighRes,Camera,BitDepth,12

# Preset: LowRes
ConfigGroup,Camera,LowRes,Camera,Binning,4
ConfigGroup,Camera,HighRes,Camera,BitDepth,8
```

Reply:
No exactly. I do think that people should do that, but I'm saying that to account for the times that that they don't we should check all the presets to get the list of all devices.

Reply:
I've added the `_check_if_presets_have_same_props` method to check if the presets have the same properties.